.github/workflows/forward-merge.yml

```diff
@@ -121,12 +163,17 @@ jobs:
         uses: ad-m/github-push-action@master
         with:
           github_token: ${{ secrets.GH_RW_TOKEN }}
-          branch: refs/heads/${{ steps.extract-next-branch.outputs.branch }}
+          branch: refs/heads/${{ needs.get-branch-names.outputs.next-branch }}
 
+  ## If we get here, it means the branches are not fast-forward mergeable, OR the merge commit
+  ## failed verification. We will need manual intervention, so we'll create a pull request to
+  ## be verified by a person.
   make-pull-request:
     runs-on: ubuntu-latest
-    if: failure()
-    needs: ['verify-merge']
+    # Run only if the verify-merge job failed. If the test-ff-only fails, but verify-merge passes, we should skip
+    if: failure() && needs.verify-merge.result == 'failure'
```

Comment:
We want to create a merge pull request only if the merge verification failed, not if the fast-forward attempt failed.